src/common/base-store.ts

```diff
@@ -19,7 +19,7 @@ export interface BaseStoreParams<T = any> extends ConfOptions<T> {
  * Note: T should only contain base JSON serializable types.
  */
 export abstract class BaseStore<T = any> extends Singleton {
-  protected storeConfig: Config<T>;
+  protected storeConfig?: Config<T>;
```

Comment:
Why are we modifying `BaseStore`?

Reply:
Because one of the errors I encountered was due to a call to `onModelChange` in the `user-store.test.ts` happening after the test suite was finished (because we don't handle rejected promises very well) and since in that test `isLoaded` was never true the function `enableSync` wasn't called so `storeConfig` was never set.

These changes merely make `BaseStore` more robust by marking the types that get late initialized as such.